src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -4,16 +4,21 @@
  *--------------------------------------------------------------------------------------------*/
 import React from 'react';
 import cx from 'classnames';
-import { StylingProps } from '../utils/props';
+import { CommonProps } from '../utils/props';
 import { useTheme } from '../utils/hooks/useTheme';
+import { useOverflow } from '../utils/hooks/useOverflow';
 import '@itwin/itwinui-css/css/button.css';
 
 export type ButtonGroupProps = {
   /**
    * Buttons in the ButtonGroup.
    */
   children: React.ReactNode;
-} & StylingProps;
+  /**
+   * If true, the ButtonGroup will collapse the overflowing Buttons into a menu.
+   */
+  responsive?: boolean;
```

Comment:
We might also need some callback in order for users to be able to show hidden items in a dropdown menu. Or should we handle that?

Reply:
Oh this is not complete. I wanted to fix the hook/test first.

I plan on adding:
- "..." as the last icon
- excessIconProps
- dropdownItems

Reply:
ok, all of the above are now supported using the `overflowButton` prop.